temporalio/activity.py

```diff
@@ -427,9 +427,14 @@ class LoggerAdapter(logging.LoggerAdapter):
         activity_info_on_message: Boolean for whether a string representation of
             a dict of some activity info will be appended to each message.
             Default is True.
-        activity_info_on_extra: Boolean for whether an ``activity_info`` value
-            will be added to the ``extra`` dictionary, making it present on the
-            ``LogRecord.__dict__`` for use by others.
+        activity_info_on_extra: Boolean for whether a ``temporal_activity``
+            dictionary value will be added to the ``extra`` dictionary with some
+            activity info, making it present on the ``LogRecord.__dict__`` for
+            use by others. Default is True.
+        full_activity_info_on_extra: Boolean for whether an ``activity_info``
+            value will be added to the ``extra`` dictionary with the entire
+            activity info, making it present on the ``LogRecord.__dict__`` for
+            use by others. Default is False.
```

Comment:
I don't think the term `full` is really clear. Maybe `activity_info_data_class_on_extra` and clarify this may not be the best option for indexing? I'd also consider deprecating this eventually, no need to have both versions long term, this is just for a transition period.

Reply:
> I don't think the term full is really clear. Maybe activity_info_data_class_on_extra and clarify this may not be the best option for indexing

I considered removing "full" but it clashed with the other. "activity info" literally refers to `activity.Info`, that it's a dataclass shouldn't apply.

> I'd also consider deprecating this eventually, no need to have both versions long term, this is just for a transition period.

Considered this, but it's off by default and harmless for the most part, so no reason not to leave it.